### ISSUE TITLE
Resize(antialias=True) blurs masks in AugmentationSequential, creating values that are not labels

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4479]
`Resize(..., antialias=True)` inside `AugmentationSequential` applies the anti-aliasing Gaussian blur to masks before the nearest-neighbour downscale. Masks then contain values that are not labels, integer masks lose and invent labels, and boolean masks change coverage (main `a7e2db6b`, torch 2.14.0, CPU):

```python
import torch
import kornia.augmentation as K

yy, xx = torch.meshgrid(torch.arange(12), torch.arange(16), indexing="ij")
blocks = ((yy // 2) + (xx // 2)) % 2   # 2x2 blocks, so a plain nearest 2x downscale keeps them exactly
for name, mask in (("float {2, 3}", (2 + blocks).float()), ("int64 {2, 3}", 2 + blocks), ("bool", blocks.bool())):
    for antialias in (False, True):
        seq = K.AugmentationSequential(K.Resize((6, 8), antialias=antialias), data_keys=["input", "mask"])
        out = seq(torch.rand(1, 3, 12, 16), mask[None, None])[1]
```

```
antialias=False float {2, 3} values=[2.0, 3.0]                          mean=2.500
antialias=True  float {2, 3} values=[2.0, 2.1065, 2.1903, 2.8097, 2.8935] mean=2.500
antialias=False int64 {2, 3} values=[2.0, 3.0]                          mean=2.500
antialias=True  int64 {2, 3} values=[1.0, 2.0]                          mean=1.979
antialias=False bool         values=[0.0, 1.0]                          mean=0.500
antialias=True  bool         values=[0.0, 1.0]                          mean=0.979
```

**Cause.** The mask path (`GeometricAugmentationBase2D.apply_transform_mask`) forces `resample` to nearest but reuses the image `flags`. `Resize.apply_transform` (`_2d/geometric/resize.py:104-112`) passes `antialias=flags["antialias"]` to `resize`, so the blur runs on the mask too.

**Expected.** Masks keep their label set under `Resize`. Anti-aliasing is image-only, for example by clearing `antialias` on the mask path the way `resample` is overridden.

Found while reviewing #4452, whose new Convention text says masks are resampled "without introducing intermediate labels" on every 2D geometric augmentation.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by Hrithikcrick** [comment]
Hi,

I'd like to work on this issue.

I'll trace the mask path through `AugmentationSequential` and `Resize`, ensure antialiasing remains image-only so masks preserve their label set, and add focused regression coverage for float, integer, and boolean masks.

I'll keep the fix scoped to this behavior and verify that normal image antialiasing remains unchanged.

Thanks!
